### PR TITLE
fix: Check HAVE_STDINT_H or HAVE_INTTYPES_H for older MSVC versions

### LINKED ISSUE #197: INTTYPES_FORMAT check fails for MSVC 15
https://github.com/ruslo/hunter/pull/601
https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1199/job/ah5vowfm7e1fxvqu

```
-- [polly] Used toolchain: Visual Studio 9 2008
2>-- The CXX compiler identification is MSVC 15.0.30729.1
2>-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 9.0/VC/bin/cl.exe
2>-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 9.0/VC/bin/cl.exe -- works
2>-- Detecting CXX compile features
2>-- Detecting CXX compile features - done
2>-- The C compiler identification is MSVC 15.0.30729.1
2>-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 9.0/VC/bin/cl.exe
2>-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 9.0/VC/bin/cl.exe -- works
2>-- Check size of __int32
2>-- Check size of __int32 - failed
2>-- Check size of uint32_t
2>-- Check size of uint32_t - failed
2>-- Check size of u_int32_t
2>-- Check size of u_int32_t - failed
2>CMake Error at CMakeLists.txt:273 (message):
2>  Do not know how to define a 32-bit integer quantity on your system! Neither
2>  uint32_t, u_int32_t, nor __int32 seem to be available.  Set
2>  [GFLAGS_]INTTYPES_FORMAT to either C99, BSD, or VC7 and try again.
2>-- Configuring incomplete, errors occurred!
2>See also "C:/projects/hunter/_testing/Hunter/_Base/56b2376/07934d6/08dcda6/Build/gflags/Build/gflags-Debug-prefix/src/gflags-Debug-build/CMakeFiles/CMakeOutput.log".
```

### PR DESCRIPTION
**Pull request by schuhschuh**
Closes #197.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,43 +1,65 @@
...
-clone_folder: c:\projects\gflags
-
-matrix:
-  fast_finish: true
+environment:
+  matrix:
+  - Toolset: v140
+  - Toolset: v120
+  - Toolset: v110
+  - Toolset: v100
```
Comment (jakirkham) [inline appveyor.yml]: Would be nice to have v90/VS 2008 as well.

Comment (schuhschuh) [inline appveyor.yml]: On it, was so far just figuring out how to have multiple toolchains with AppVeyor...

Comment (jakirkham) [inline appveyor.yml]: No worries. Cool, thanks.

**Comment by jakirkham** [comment]
I think 64-bit VS 2010 isn't working. Not sure if 64-bit support was included in that compiler (I think it wasn't). The same story exists with VS 2008. In conda-forge we work around this issue with VS 2008 using a [hack]( https://github.com/conda-forge/vs2008_express_vc_python_patch-feedstock/blob/master/recipe/vs2008_patch/setup_x64.bat ) to add 64-bit support. A similar thing is likely needed with VS 2010 for 64-bit. That said, I'm not sure if it is worth including here or if those builds shouldn't just be skipped.

**Comment by schuhschuh** [comment]
Thanks, just been looking it. I only copied the config from the gtest project, where they excluded it for 2010 already.

**Comment by jakirkham** [comment]
So AppVeyor allows [exclusions of combinations]( https://www.appveyor.com/docs/build-configuration/#exclude-configuration-from-the-matrix ) from the matrix, which may help here.

**Comment by schuhschuh** [comment]
Alright, seems to be all fine now.

File: appveyor.yml [hunk 2]
```diff
@@ -1,43 +1,77 @@
...
-    -D GFLAGS_BUILD_TESTING=ON
-    -D GFLAGS_BUILD_SHARED_LIBS=ON
-    -D GFLAGS_BUILD_STATIC_LIBS=ON
-    ..
-  - cmake --build . --config %configuration%
+- ps: |
+    if ((($env:Toolset -eq "v90") -or ($env:Toolset -eq "v100")) -and ($env:PLATFORM -eq "x64"))
+    {
+        return
+    }
```
Comment (jakirkham) [inline appveyor.yml]: Would drop this check.

File: appveyor.yml [hunk 3]
```diff
@@ -1,43 +1,77 @@
...
+        throw "Exec: $ErrorMessage"
+    }
 
 test_script:
-  - ctest -C %configuration%
+- ps: |
+    if ((($env:Toolset -eq "v90") -or ($env:Toolset -eq "v100")) -and ($env:PLATFORM -eq "x64"))
+    {
+        return
+    }
```
Comment (jakirkham) [inline appveyor.yml]: Also would drop this one too.

Comment (schuhschuh) [inline appveyor.yml]: Well, I thought it doesn't hurt as it's not the most important file anyway... but alright

**Comment by jakirkham** [comment]
Indeed. Looks good. Suggested cleanup of some now unneeded portions.

File: appveyor.yml (hunk 3 above)
Comment (jakirkham) [inline appveyor.yml]: Was mainly suggesting to keep things easy to read. But yes you're right that there is no functional issue having it.

File: appveyor.yml (hunk 1 above)
Comment (jakirkham) [inline appveyor.yml]: note: Done below.

File: appveyor.yml (hunk 3 above)
Comment (schuhschuh) [inline appveyor.yml]: Of course, and makes sense. Just didn't mean to bother after all the iterations and though to myself maybe a good reference if at another time I need to figure out how to have such condition in PowerShell... which I never work with. But then again, for that we got the internet when needed :-).

**Pull request merged**